src/csv_arrow_ingest.cpp

```diff
@@ -0,0 +1,98 @@
+#include <arrow/buffer.h>
+#include <arrow/io/api.h>
+#include <arrow/util/compression.h>
+#include <csv_arrow_ingest.hpp>
+#include <decryption.hpp>
+
+arrow::csv::ConvertOptions
+get_arrow_convert_options(std::vector<std::string> &utf8_columns) {
+  auto convert_options = arrow::csv::ConvertOptions::Defaults();
+  if (!utf8_columns.empty()) {
```

Comment:
nit: this `if` isn't really needed - if `utf8_columns` is empty we won't enter the loop anyway :-)

Reply:
a nice side benefit of using references. done